### PR TITLE
fix thread interrupted exception reset flag: solves #23577

### PR DESCRIPTION
**Pull request by jialigit**
# Description
solves [ thread interrupted exception reset flag ](https://sonarcloud.io/project/issues?resolved=false&types=BUG&id=stliu_azure-sdk-for-java&open=AXtAG3Fr00FAjxsg1BZV) for #23577 of sonar scan.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/graph/AadGraphClientTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        } catch (InterruptedException e) {
+            assertFalse(Thread.currentThread().isInterrupted());
+            // Restore interrupted state...
+            Thread.currentThread().interrupt();
+            isInterrupted = true;
+        }
+
+        assertTrue(Thread.currentThread().isInterrupted());
+        assertTrue(isInterrupted);
+    }
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/graph/AadGraphClientTest.java:39]: Is it possible that we can use two threads to test the case, instead of using the same thread?

Comment (jialigit) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/graph/AadGraphClientTest.java:39]: I write this case, but I find it not looks good and time wasting and counter- pruductive for testing. 
    refer:[gist](https://gist.github.com/jialigit/ea5f8319c251e12437288e3d08b4accf)

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/webapi/AadOboOAuth2AuthorizedClientProvider.java [hunk 2]
```diff
@@ -144,15 +144,19 @@ private <T extends OAuth2AuthorizedClient> T getOboAuthorizedClient(ClientRegist
...
-                .filter(StringUtils::hasText)
-                .ifPresent(this::replyForbiddenWithWwwAuthenticateHeader);
-            LOGGER.error("Failed to load authorized client.", exception);
-        } catch (InterruptedException | ParseException exception) {
+                    .map(Throwable::getCause)
+                    .filter(e -> e instanceof MsalInteractionRequiredException)
+                    .map(e -> (MsalInteractionRequiredException) e)
+                    .map(MsalServiceException::claims)
+                    .filter(StringUtils::hasText)
+                    .ifPresent(this::replyForbiddenWithWwwAuthenticateHeader);
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/webapi/AadOboOAuth2AuthorizedClientProvider.java:152]: Do not do unnecessary change.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/graph/AadGraphClientTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+package com.azure.spring.cloud.autoconfigure.implementation.graph;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class AadGraphClientTest {
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/graph/AadGraphClientTest.java:13]: Seems this test has no relationship with `AadGraphClient`?

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/graph/AadGraphClientTest.java (hunk 3 above)
Comment (jialigit) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/graph/AadGraphClientTest.java:13]: yes, this case tests the status of the thread interrupt. And mocking AadGraphClient is counter-productive.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/webapi/AadOboOAuth2AuthorizedClientProvider.java (hunk 2 above)
Comment (jialigit) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/webapi/AadOboOAuth2AuthorizedClientProvider.java]: ok,  revert this lines.

**Review by chenrujun — APPROVED** [review]
LGTM.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/graph/AadGraphClientTest.java (hunk 1 above)
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/implementation/graph/AadGraphClientTest.java:39]: I don't think this test case here cover the most common multi-thread case.

**Review by saragluna — APPROVED** [review]
LGTM.
